app/models/post.rb

```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  validates :title, presence: true
+  validates :text, presence: true
+  validates :comment_counter, presence: false
+  validates :likes_counter, presence: false
+
+  has_many :comments
+  has_many :likes
+  belongs_to :user
+  # retrieve all users who have like a post
+  has_many :likers, through: :likes, source: :user
+  # retrieve all users who have commented on a post
+  has_many :commenters, through: :comments, source: :user
+
+  def update_comment_counter
+    update(comment_counter: comments.count)
+  end
+
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
+end
```

Comment:
- [ ] According to the project requirements, this method to update the comments counter should be in the `Comment` model. Please rectify this 👍 
**HINT:** In the `Comment` model, you can take advantage of the relationship between `post` and `comments` like this 👇
```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```
- [ ] In like manner, the `#update_likes_counter` method would be better placed in the `Like` model. Kindly implement this requirement.